src/web/methods/checkers.methods.ts

```diff
@@ -0,0 +1,272 @@
+    });
+  } else if (assert === 'is not') {
+    return currentPage.waitForVisibilityOf(elementName).then(() => {
+      return checkMatcher(pageElement, matcher)
+        .catch(() => Promise.resolve(false))
+        .then(result => (result ? Promise.reject() : Promise.resolve()));
+    });
+  }
+};
+
+export const checkElementWithMatcher = (
+  currentPage: BasePage,
+  assert: string,
+  elementName: string,
+  matcher: string
+) => {
+  const pageElement = currentPage.getElement(elementName);
+  const checkMatcher = (ele, val) => {
+    return matchers.match(ele, variableStore.replaceTextVariables(`f:${val}`));
+  };
+
+  if (assert === 'is') {
+    return currentPage.waitForVisibilityOf(elementName).then(() => {
+      return checkMatcher(pageElement, matcher).then(matcherResult => expect(matcherResult).toBe(true));
+    });
+  } else if (assert === 'is not') {
+    return currentPage.waitForVisibilityOf(elementName).then(() => {
+      return checkMatcher(pageElement, matcher)
+        .catch(() => Promise.resolve(false))
+        .then(result => (result ? Promise.reject() : Promise.resolve()));
+    });
+  }
+};
+
+export const checkElementContainingText = (
+  currentPage: BasePage,
+  assert: string,
+  elementName: string,
+  value: string
+) => {
+  const pageElement = currentPage.getElement(elementName);
+  const checkMatcher = (ele, val) => {
+    return matchers.match(ele, variableStore.replaceTextVariables(`t:${val}`));
+  };
+
+  if (assert === 'is') {
+    return currentPage.waitForVisibilityOf(elementName).then(() => {
+      return checkMatcher(pageElement, value);
+    });
+  } else if (assert === 'is not') {
+    return currentPage.waitForVisibilityOf(elementName).then(() => {
+      return checkMatcher(pageElement, value)
+        .catch(() => Promise.resolve(false))
+        .then(result => (result ? Promise.reject() : Promise.resolve()));
+    });
+  }
+};
+
+export const checkFollowingElementsInTable = (currentPage: BasePage, table: string, data: TableDefinition) => {
+  const allElements = currentPage.getElements(table);
+  const hashes = data.hashes();
+  return currentPage.waitForVisibilityOf(table).then(() => {
+    return this.checkNumberOfElements(currentPage, `equal ${hashes.length}`, table).then(() => {
+      const promises = [];
+
+      return allElements
+        .each((element, index) => {
+          const hash = hashes[index];
+
+          for (const prop in hash) {
+            if (hash.hasOwnProperty(prop)) {
+              const propValue = hash[prop];
+
+              promises.push(
+                matchers.match(
+                  element.element(currentPage.getElement(prop).locator()),
+                  valueToTextTransformer.transform(propValue)
+                )
+              );
+            }
+          }
+        })
+        .then(() => Promise.all(promises));
+    });
+  });
+};
+
+export const checkFollowingElementsForElement = (
```

Comment:
Improve the names of those methods - I don't get what this method is used for

Reply:
@tgorskievo @joszo suggested to remove check from this names as we use `methods.checkers.nameOfCheck`

Reply:
I dont see a reason to remove "check". I suppose this is a method to check if array of items is available under specific selector right?

Reply:
yes @sethii, I was using step definitions to name those export functions so they, in my opinion, are straight forward